src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientSpec.cs

```diff
@@ -141,194 +273,350 @@ private RoleName GetRoleName(Address address)
+                    TestConductor.Blackhole(_config.Client, receptionistRoleName, ThrottleTransportAdapter.Direction.Both).Wait();
+                    c.Tell(new ClusterClient.Send("/user/service2", "ping", localAffinity: true));
                     // if we would use remote watch the failure detector would trigger and
                     // connection quarantined
-                    ExpectNoMsg(TimeSpan.FromSeconds(5));
+                    ExpectNoMsg(5.Seconds());
 
-                    TestConductor.PassThrough(_client, receptionistRoleName, ThrottleTransportAdapter.Direction.Both).Wait();
+                    TestConductor.PassThrough(_config.Client, receptionistRoleName, ThrottleTransportAdapter.Direction.Both).Wait();
 
-                    var expectedAddress = Node(receptionistRoleName).Address;
+                    var expectedAddress = GetAddress(receptionistRoleName);
                     AwaitAssert(() =>
                     {
-                        c.Tell(new Client.ClusterClient.Send("/user/service2", "bonjour3", localAffinity: true));
-                        ExpectMsg("bonjour3-ack", TimeSpan.FromSeconds(1));
-                        var lastSenderAddress2 = LastSender.Path.Address;
-                        Assert.Equal(expectedAddress, lastSenderAddress2);
+                        var probe = CreateTestProbe();
+                        c.Tell(new ClusterClient.Send("/user/service2", "bonjour3", localAffinity: true), probe.Ref);
+                        var reply2 = probe.ExpectMsg<ClusterClientSpecConfig.Reply>(1.Seconds());
+                        reply2.Msg.Should().Be("bonjour3-ack");
+                        reply2.Node.Should().Be(expectedAddress);
                     });
                     Sys.Stop(c);
+                }, _config.Client);
 
-                }, _client);
                 EnterBarrier("after-5");
             });
         }
+
+        public void ClusterClient_must_reestablish_connection_to_receptionist_after_server_restart()
+        {
+            Within(30.Seconds(), () =>
+            {
+                RunOn(() =>
+                {
+                    _remainingServerRoleNames.Count.Should().Be(1);
+                    var remainingContacts = _remainingServerRoleNames.Select(r => Node(r) / "system" / "receptionist").ToList();
+                    var c = Sys.ActorOf(ClusterClient.Props(ClusterClientSettings.Create(Sys).WithInitialContacts(remainingContacts)), "client4");
+                    c.Tell(new ClusterClient.Send("/user/service2", "bonjour4", localAffinity: true));
+                    var reply = ExpectMsg<ClusterClientSpecConfig.Reply>(10.Seconds());
+                    reply.Msg.Should().Be("bonjour4-ack");
+                    reply.Node.Should().Be(remainingContacts.First().Address);
+
+                    var logSource = $"{(Sys as ExtendedActorSystem).Provider.DefaultAddress}/user/client4";
```

Comment:
This test test is not pass, because we are waiting for `Info` message here
```
EventFilter.Info(start: "Connected to", source: 'akka.trttl.gremlin.tcp://MultiNodeClusterSpec@localhost:57138/user/client4')
```
but it happens in this LogSource
```
akka.trttl.gremlin.tcp://MultiNodeClusterSpec@localhost:57138/user/client4
```